### PR TITLE
Snippet Injection Record

### PR DESCRIPTION
**Pull request by siyuniums**
Try to open a PR

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java [hunk 1]
```diff
@@ -102,6 +106,16 @@ public class SecondEntryPoint implements AutoConfigurationCustomizerProvider {
   @Nullable private static BatchSpanProcessor batchSpanProcessor;
   @Nullable private static MetricReader metricReader;
 
+  public String readFile(String resourceName) throws IOException {
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java:109]:
```suggestion
  private static String readFile(String resourceName) throws IOException {
```

**Review by trask — APPROVED** [review]
in a follow-up, let's add a json configuration option under "preview" to enable snippet injection (since we don't want it enabled by default)

**Comment by jeanbisutti** [comment]
@trask @siyuniums The tests can't run: `Could not find io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.16.0-alpha-SNAPSHOT.`

**Comment by siyuniums** [comment]
> @trask @siyuniums The tests can't run: `Could not find io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.16.0-alpha-SNAPSHOT.`

Hi Jean, thanks for the reviewing. This PR is only for a code record. The snapshot is a new local version open telemetry which have the snippet injection enabled.

**Comment by trask** [comment]
> Hi Jean, thanks for the reviewing. This PR is only for a code record. The snapshot is a new local version open telemetry which have the snippet injection enabled.

this PR is targeted to the `snippet-injection` branch where we'll track these changes until we're able to get the upstream changes landed and then we'll send PR to propose merging the `snippet-injection` branch to `main`

**Pull request closed without merging**